From 0d718a4b68fd1f74c6cf96634d7d862a8f1e7465 Mon Sep 17 00:00:00 2001
From: xuxiang <xuxiang@eswincomputing.com>
Date: Wed, 22 May 2024 16:17:24 +0800
Subject: [PATCH 047/109] WIN2030-15279:fix:i2c status not ready

Changelogs:
1. mcpu reset causes mcu i2c state one field
2. Optimize som_rst_feedback_process
3. Optimize uart, i2cinit functions,

Change-Id: Iddc3736f261fd7391ee929a2e2d95f34b9d2445a
---
 Core/Src/hf_gpio_process.c  | 14 ++------------
 Core/Src/hf_power_process.c |  2 ++
 2 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/Core/Src/hf_gpio_process.c b/Core/Src/hf_gpio_process.c
index b09a51e..2a53ed3 100644
--- a/Core/Src/hf_gpio_process.c
+++ b/Core/Src/hf_gpio_process.c
@@ -144,19 +144,9 @@ static void mcu_reset_som_process(void)
 static void som_rst_feedback_process(void)
 {
 	printf("%s %d som reset\n", __func__, __LINE__);
-	GPIO_InitTypeDef GPIO_InitStruct = {0};
-	GPIO_InitStruct.Pin = MCU_RESET_SOM_N_Pin;
-	GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
-	GPIO_InitStruct.Pull = GPIO_NOPULL;
-	GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
-	HAL_GPIO_Init(MCU_RESET_SOM_N_GPIO_Port, &GPIO_InitStruct);
-	HAL_GPIO_WritePin(MCU_RESET_SOM_N_GPIO_Port, MCU_RESET_SOM_N_Pin, GPIO_PIN_RESET);
+	som_reset_control(pdTRUE);
 	osDelay(2);
-	GPIO_InitStruct.Pin = MCU_RESET_SOM_N_Pin;
-	GPIO_InitStruct.Mode = GPIO_MODE_IT_FALLING;
-	GPIO_InitStruct.Pull = GPIO_NOPULL;
-	GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
-	HAL_GPIO_Init(MCU_RESET_SOM_N_GPIO_Port, &GPIO_InitStruct);
+	som_reset_control(pdFALSE);
 }
 
 void hf_gpio_task(void *parameter)
diff --git a/Core/Src/hf_power_process.c b/Core/Src/hf_power_process.c
index 066d1a0..3f7991e 100644
--- a/Core/Src/hf_power_process.c
+++ b/Core/Src/hf_power_process.c
@@ -343,6 +343,7 @@ void som_reset_control(uint8_t reset)
 		GPIO_InitStruct.Pull = GPIO_NOPULL;
 		GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
 		HAL_GPIO_Init(MCU_RESET_SOM_N_GPIO_Port, &GPIO_InitStruct);
+		i2c_deinit(I2C3);
 		uart_deinit(UART4);
 		uart_deinit(USART6);
 		SPI2_FLASH_CS_LOW();
@@ -361,6 +362,7 @@ void som_reset_control(uint8_t reset)
 		HAL_GPIO_Init(MCU_RESET_SOM_N_GPIO_Port, &GPIO_InitStruct);
 		uart_init(UART4);
 		uart_init(USART6);
+		i2c_init(I2C3);
 		SPI2_FLASH_CS_HIGH();
 	}
 }
-- 
2.25.1

